otherlibs/dune-site/test/install-subst.t

```diff
@@ -0,0 +1,56 @@
+This checks that generated sites files correctly substituted with `dune install`.
+
+See #10317.
+
+  $ cat > dune-project << EOF
+  > (lang dune 3.0)
+  > (using dune_site 0.1)
+  > (package
+  >  (name a)
+  >  (sites (share data)))
+  > EOF
+
+  $ cat > dune << EOF
+  > (library
+  >  (public_name a)
+  >  (libraries dune-site))
+  > (generate_sites_module
+  >  (module s)
+  >  (sites a))
+  > EOF
+
+  $ dune build
+
+  $ has_placeholder() {
+  >   name=$1;
+  >   if grep -q DUNE_PLACEHOLDER "$name"; then
+  >     printf "%-30s has placeholders\n" "$name"
+  >   else
+  >     printf "%-30s has no placeholders\n" "$name"
```

Comment:
Maybe it should not print anything in the "success" case, that way files that are not that relevant are not included in the output thus making the test less fragile.